internal/tests/api/wallet_test.go

```diff
@@ -27,89 +26,33 @@ var (
 
+func TestWalletRecover_Success(t *testing.T) {
 	// Run the round-trip test
 	derivationPath := string(wallet.DerivationPath_Default)
 	index := uint64(0)
 	response, err := hdNode.GetApiClient().Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
 	require.NoError(t, err)
 	t.Log("Recover called")
 
-	// Check the response
-	require.Equal(t, expectedWalletAddress, response.Data.AccountAddress)
-	t.Log("Received correct wallet address")
-}
-
-func TestWalletRecover_WrongIndex(t *testing.T) {
 	// Take a snapshot, revert at the end
-	snapshotName, err := testMgr.CreateCustomSnapshot(hdtesting.Service_Filesystem)
-	if err != nil {
-		fail("Error creating custom snapshot: %v", err)
-	}
-	defer wallet_cleanup(snapshotName)
-
-	// Run the round-trip test
-	derivationPath := string(wallet.DerivationPath_Default)
-	index := uint64(1)
-	response, err := hdNode.GetApiClient().Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
-	require.NoError(t, err)
-	t.Log("Recover called")
-
-	// Check the response
-	require.NotEqual(t, expectedWalletAddress, response.Data.AccountAddress)
-	t.Logf("Wallet address doesn't match as expected (expected %s, got %s)", expectedWalletAddress.Hex(), response.Data.AccountAddress.Hex())
-}
-
-func TestWalletRecover_WrongDerivationPath(t *testing.T) {
-	snapshotName, err := testMgr.CreateCustomSnapshot(hdtesting.Service_Filesystem)
+	walletTestWalletRecoveredSnapshot, err = testMgr.CreateSnapshot()
 	if err != nil {
 		fail("Error creating custom snapshot: %v", err)
 	}
-	defer wallet_cleanup(snapshotName)
-
-	// Run the round-trip test
-	derivationPath := string(wallet.DerivationPath_LedgerLive)
-	index := uint64(0)
-	response, err := hdNode.GetApiClient().Wallet.Recover(&derivationPath, keys.DefaultMnemonic, &index, goodPassword, true)
-	require.NoError(t, err)
-	t.Log("Recover called")
+	defer wallet_cleanup(baseSnapshot, TestMain_BaseSnapshot, t)
 
 	// Check the response
-	require.NotEqual(t, expectedWalletAddress, response.Data.AccountAddress)
-	t.Logf("Wallet address doesn't match as expected (expected %s, got %s)", expectedWalletAddress.Hex(), response.Data.AccountAddress.Hex())
-}
-
-func TestWalletStatus_NotLoaded(t *testing.T) {
-	apiClient := hdNode.GetApiClient()
-	response, err := apiClient.Wallet.Status()
-	require.NoError(t, err)
-	t.Log("Status called")
-
-	require.Equal(t, emptyWalletAddress, response.Data.WalletStatus.Address.NodeAddress)
-	require.False(t, response.Data.WalletStatus.Address.HasAddress)
-
-	require.Equal(t, wallet.WalletType(""), response.Data.WalletStatus.Wallet.Type)
-	require.False(t, response.Data.WalletStatus.Wallet.IsLoaded)
-	require.False(t, response.Data.WalletStatus.Wallet.IsOnDisk)
-	require.Equal(t, emptyWalletAddress, response.Data.WalletStatus.Wallet.WalletAddress)
-
-	t.Log("Received correct wallet status")
+	require.Equal(t, expectedWalletAddress, response.Data.AccountAddress)
```

Comment:
This should come *before* the wallet snapshot; snapshotting a setup that failed its condition checks doesn't make any sense.